### ISSUE TITLE
Add option to disable multiplication of rows on selection

### ISSUE THREAD

**Issue by nichtich** [deutsche-nationalbibliothek/pica-rs#490]
[select command](https://github.com/deutsche-nationalbibliothek/pica-rs#select) multiplies rows for repeatable subfields. This results in large result sets and requires postprocessing if this multiplication is not wanted. I would like an option to disable multiplication of rows. Repeatable subfield values should be concatenated with a delimiter instead, e.g.

~~~
$ pica select -s "003@.0,012A/*{a,b,c}" DUMP.dat.gz
123456789X,a,b,c
123456789X,d,e,f

$ pica select -s "003@.0,012A/*{a,b,c}" -m "|" DUMP.dat.gz
123456789X,a|d,b|e,c|f
~~~

**Comment by niko2342** [comment]
This seems to be a good extension. It's not easy to implement, but a refactoring of the select code is planned and it could be a good time to implement this.

Questions:

- All three subfield values of the first command are different. This is the case when the field is repeated and each subfield contains only one value. Does the example contains an error? The feature request is about repeated subfield values and not about repeated fields, right?
- What does `-m` stands for? (I would prefer an long option like `--delimiter` or `--separator`)
- It may happen that the subfield value contains the specified delimiter, which can cause an error in an subsequent processing step. How to handle this problem?

CC: @mfakaehler , @mschumac

**Comment by nichtich** [comment]
> The feature request is about repeated subfield values and not about repeated fields, right?

I expect the two records 

    012A $x1$x2

and

    012A $x1
    012A $x2

to give the same selection results, e.g `1+2` with delimiter `+`.

> What does `-m` stands for? (I would prefer an long option like `--delimiter` or `--separator`)

I prefer to have both but the name and letter are up to you.

> It may happen that the subfield value contains the specified delimiter, which can cause an error in an subsequent processing step. How to handle this problem?

I'd better not handle this case in a special way but keep it as task of the user to choose an approriate delimiter, depending on use case. The empty string should also be allowed.

**Comment by a-wendler** [comment]
This feature would be awesome. As far as I can see, the suggestion solution looks great. Choosing the delimiter yourself is highly wanted.

**Comment by nichtich** [comment]
Given that "delimiter" is already refered to as column delimiter in TSV/CSV, I'd better use the term "separator".

**Comment by niko2342** [comment]
I've created a proposal how this could be implemented, feel free to add your comments! see #583
